### DISCUSSION TITLE
Drop requirement on coverage build

### DISCUSSION THREAD

**Discussion by j0k3r** [serverless-heaven/serverless-webpack#1558] [Team Posts]
Hello @HyperBrain,

I hope you're fine.
Can you remove the requirement for the `coverage/coveralls` build?
I would arg against in the past to ensure the coverage doesn't drop too much but we have a case today which switch my POV.

One Serverless maintainer propose an update to the plugin about the upcoming v3 version of the framework. We can't test multiple version of the framework against the test suite so the coverage will drop anyway. Even with the addition of a custom build with Serverless pre-v3.
https://github.com/serverless-heaven/serverless-webpack/pull/1013

Thanks!

**Comment by vicary** [comment]
A gentle ping to @HyperBrain.

**Comment by HyperBrain** [comment]
Hi guys,

sorry for the late reply. I will do that tomorrow evening.

**Comment by HyperBrain** [comment]
I disabled the coveralls "no drop" requirement now in the coveralls settings.
However, coveralls is still run to be aware of the overall coverage. Please let me know if the issues are now fixed.
As coverage control ensured a good quality in the past (especially in the times where I was nearly the only one maintaining the project) we/you should consider activating it again at some point in the future to make sure that people always add proper tests for their submitted code.
Just saw - we still have an amazing 94% of coverage ;)

Additionally I want to take the time to say a big "thank you". Without your help the project would have been died due to my limited time in the past year.

Cheers 
Frank

**Comment by j0k3r** [comment]
I agree with you.
That’s why we try to add e2e test to better test the missing part. But we are failing for now.
Thanks!

**Comment by j0k3r** [comment]
Hey @HyperBrain it looks like the requirement on GitHub is still here and we can't merge when the coverage drop ...
See https://github.com/serverless-heaven/serverless-webpack/pull/1018

**Comment by HyperBrain** [comment]
Ok. Then I need to check again today, where the remaining settings can be
found. In the worst case I would just disable coveralls for the moment and
enable it again as soon as possible.

**Comment by j0k3r** [comment]
If you can't find a solution on Coveralls side, maybe you can just disable the requirement on GitHub side.

**Comment by miguel-a-calles-mba** [comment]
It seems the code coverage dropped because there are no unit tests for this new function.
https://github.com/serverless-heaven/serverless-webpack/pull/1018/files#diff-bd3dad1576553f0425c8ed3f0aedb28cb66035b78e9461c300c4fc41871dffc0R34

**Comment by HyperBrain** [comment]
There is no setting on GitHub side and in Coveralls it is disabled:
![image](https://user-images.githubusercontent.com/5524702/159244720-8a6055c1-52d2-44ed-88db-74251e1b6ccb.png)
Can it be that it needs to be run but ignored in the travis build file?

**Comment by j0k3r** [comment]
In fact, on Coveralls side you can leave the value to mark the build as fail if the drop is too important.

What you must update on GitHub side is the branch protection rules ([here](https://github.com/serverless-heaven/serverless-webpack/settings/branches)) and remove Coveralls in the section _Require status checks to pass before merging_ so we'll be able to merge any PR with coverage drops because GitHub won't block it, only notify us.

**Comment by HyperBrain** [comment]
The branch protection rule is changed. I removed the necessity for coveralls.

**Comment by j0k3r** [comment]
Thanks!
